src/hotspot/cpu/s390/c1_Defs_s390.hpp

```diff
@@ -74,7 +74,12 @@ enum {
 
 // the number of stack required by ArrayCopyStub
 enum {
-  pd_arraycopystub_reserved_argument_area_size = 2
+  pd_arraycopystub_reserved_argument_area_size = 4
```

Comment:
Do you know the root cause of failure? Maybe only the modification here are needed. You can try.

Reply:
this isn't fixing the issue. Beside that I had to make changes in `src/hotspot/share/c1/c1_Compilation.cpp` because `compiler/c1/KlassAccessCheckTest.java` appeared after build fix and it was failing with the same error as build.

Reply:
Sure; [KlassAccessCheckTest.log](https://github.com/openjdk/jdk/files/10895005/KlassAccessCheckTest.log)

Reply:
I want to see /home/amit/build_test/jdk/build/linux-s390x-server-fastdebug/test-support/jtreg_test_hotspot_jtreg_tier1/scratch/1/hs_err_pid901198.log

Reply:
I would like to see more detailed stack information from hs_err_pid901198.log, so to know which stub is causing the problem.

Reply:
I'm really sorry. 
[hs_err_pid901198.log](https://github.com/openjdk/jdk/files/10895097/hs_err_pid901198.log)

Reply:
Thank you. perhaps this is the difference between the S390 and other architectures, do not call store_parameter() in this method on other architectures, such as x86. 

LIR_Assembler::emit_typecheck_helper(LIR_OpTypeCheck*, Label*, Label*, Label*)+0x800  (c1_LIRAssembler_s390.cpp:2534)

Next, let's see if there is a better solution.

Reply:
https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/s390/c1_LIRAssembler_s390.cpp#L2534
Maybe you can try store klass_RInfo & k_RInfo to current sp location but not  reserve_stack space. such as:
__ z_stg(klass_RInfo, offset_in_bytes, Z_SP);

Reply:
But there are other call sites as well, what about them ? You mean to change them all ?

Reply:
I don't know much about the implementation of `emit_typecheck_helper` yet, maybe you can refer to aarch64.

Reply:
According to my understanding, C1 stack space is divided into two parts: general stack space ( see hir()->max_stack) and reserved stack space (see reserved_argument_area, for stub used).
`emit_typecheck_helper` is not stub, so shouldn't use the reserved stack.